src/lean_spec/subspecs/containers/state/state.py

```diff
@@ -440,3 +440,54 @@ def state_transition(self, block: Block, valid_signatures: bool = True) -> "Stat
             raise AssertionError("Invalid block state root")
 
         return new_state
+
+    def produce_block(self, target_slot: Slot, is_empty: bool = False) -> Block:
```

Comment:
Since this is added to the spec, when would we be using this function when we already have [`produce_block_with_signatures()`](https://github.com/leanEthereum/leanSpec/blob/3e812e0b77eebcae446ee59a8384f9fdf5b60fd6/src/lean_spec/subspecs/forkchoice/store.py#L691)?